pkg/kernel/networkservice/applyiptables4nattemplate/server.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2022 Xored Software Inc and others.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+// +build linux
+
+// Package applyiptables4nattemplate provides chain element for setup iptables nat rules
+package applyiptables4nattemplate
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/kernel"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/utils/metadata"
+)
+
+type iptablesServer struct {
+	manager IPTablesManager
+}
+
+// Option is an option pattern for applyiptables4nattemplate chain elements
+type Option func(server *iptablesServer)
+
+// WithIPTablesManager provides an option to change IPTablesManager for a chain element
+func WithIPTablesManager(iptable IPTablesManager) Option {
+	return func(server *iptablesServer) {
+		server.manager = iptable
+	}
+}
+
+// NewServer - returns a new networkservice.NetworkServiceServer that modify IPTables rules
+// by mechanism provided template on Request and rollbacks rules changes on Close
+func NewServer(options ...Option) networkservice.NetworkServiceServer {
```

Comment:
Do we really need options if dont have tests for this moment?

Reply:
I left it for the future. I agree, it is better to remove unused feature it for now. Done.